compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -24,6 +27,22 @@ pub(crate) enum AccessKind {
     Mutate,
 }
 
+#[derive(Debug)]
+pub(crate) enum PlaceAndReason<'a> {
+    // place, name
+    DeclaredImmute(String, Option<String>),
```

Comment:
It would be good if we could take the `describe_any_place` logic and put it into a `IntoDiagnosticArg` so that types like this could contain the `Place`s themselves.